controllers/session.js

```diff
@@ -0,0 +1,168 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const Joi = require('joi');
+const User = require('../models/users');
+const gravatar = require('gravatar');
+const mailgun = require('mailgun-js');
+const { v4: uuidv4 } = require('uuid');
+const nodemailer = require('nodemailer');
+
+// Konfiguracja Mailgun
+const { MAILGUN_API_KEY, MAILGUN_DOMAIN } = process.env;
+const mg = mailgun({ apiKey: MAILGUN_API_KEY, domain: MAILGUN_DOMAIN });
+
+// Walidacja użytkownika
+const userSchema = Joi.object({
+  email: Joi.string().email().required(),
+  password: Joi.string().min(6).required(),
+});
+
+// Funkcja do wysyłania e-maila weryfikacyjnego
+const sendVerificationEmail = async (userEmail, verificationToken) => {
+  const verificationUrl = `http://yourdomain.com/api/users/verify/${verificationToken}`;
+
+  const data = {
+    from: `noreply@${MAILGUN_DOMAIN}`,
+    to: userEmail,
+    subject: 'Email Verification',
+    text: `Please verify your email by clicking the following link: ${verificationUrl}`,
+  };
+
+  try {
+    await mg.messages().send(data);
+  } catch (error) {
+    console.error('Error sending verification email:', error);
```

Comment:
```suggestion
    console.error('Error sending verification email:', error);
    throw error;
```
Takie coś sprawi że w miejscu gdzie używasz tej funkcji będziesz mógł wykorzystać błąd i podjąć akcję.